### PR TITLE
Issue #6726: aligned javadoc/xdoc for SuppressWithPlainTextCommentFilter

### PR DESCRIPTION
**Pull request by pbludov**
Issue #6726 (part of #5750)

No changes in the code.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilter.java [hunk 1]
```diff
@@ -217,7 +457,7 @@ private static Suppression getNearestSuppression(List<Suppression> suppressions,
     }
 
     /** The class which represents the suppression. */
-    public static class Suppression {
+    /* package */ static class Suppression {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilter.java]: Why did this change?

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilter.java]: To hide the implementation details. But this should be `private`. I'll start an issue for this.

**Comment by romani** [comment]
@pbludov ,

TC is failured:
```
src/main/java/com/puppycrawl/tools/checkstyle/filters
SuppressWithPlainTextCommentFilter.java (1)
460: Suppression Package-visible nested class Suppression 
```

Please fix.

**Comment by pbludov** [comment]
New issue to fix internal class: https://github.com/checkstyle/checkstyle/issues/6890

**Pull request merged**